akka-docs/rst/scala/cluster-sharding.rst

```diff
@@ -274,6 +275,15 @@ using a ``Passivate``.
 Note that the state of the entries themselves will not be restored unless they have been made persistent,
 e.g. with ``akka-persistence``.
 
+Graceful Shutdown
+-----------------
+
+You can send the message ``ClusterSharding.GracefulShutdown`` message (``ClusterSharding.gracefulShutdownInstance
+in Java) to the ``ShardRegion`` actor to handoff all shards that are hosted by that ``ShardRegion`` and then the
+``ShardRegion`` actor will be stopped. You can ``watch`` the ``ShardRegion`` actor to know when it is completed.
+During this period other regions will buffer messages for those shards in the same way as when a rebalance is
+triggered by the coordinator. When the shards have been stopped the coordinator will allocate these shards elsewhere. 
```

Comment:
I don't think that matters. You would watch the ShardRegion actor, and when it terminates you would cluster.leave(cluster.selfAddress) and that will trigger the graceful failover of the coordinator singleton.

This scenario should probably be explained here (with code).